core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -119,8 +126,6 @@ object SizeEstimator extends Logging {
       }
     }
     pointerSize = if (is64bit && !isCompressedOops) 8 else 4
-    classInfos.clear()
-    classInfos.put(classOf[Object], new ClassInfo(objectSize, Nil))
```

Comment:
No need to clear() because we're using ThreadLocal WeakHashMaps so the initialisation will occur once per thread, we put the new value in and we assume the size won't be changing which is going to be the case except for running in some debug modes*. As they're in a ThreadLocal WeakHashMap the classes can still be unloaded if no longer used.

* I'm referring to JDK debug modes such as fullspeeddebug where we can change class layouts at runtime. Normal execution modes prohibit changes to the class layout during execution without the class being unloaded and reloaded which will trigger the map entry to clear and be recreated due to its weakness (so in normal cases the map will always be correct)